### PR TITLE
Refactor compiler c++ std settings

### PR DESCRIPTION
**Pull request by niranjanyardi**
b/329683364

Consolidate apple, windows host, default toolchain c++ standards settings.

### CODE REVIEW

File: starboard/build/config/BUILD.gn [hunk 1]
```diff
@@ -241,20 +240,17 @@ config("default_cpp_standard") {
...
       cflags_cc = [ "-std=c++17" ]
-    }
-
-    # These flags apply to non-modular windows platforms which includes win32, xb1.
-    if (is_host_win && !sb_is_modular) {
-      cflags = [ "/std:c++17" ]
-    }
-
-    if (is_linux && !sb_is_modular) {
+    } else if (is_host_win && !sb_is_modular) {
```
Comment (andrewsavage1) [inline starboard/build/config/BUILD.gn:243]: You can simplify a lot of these conditionals now that they're `else if`s, i.e. remove !sb_is_modular

File: starboard/build/config/BUILD.gn [hunk 2]
```diff
@@ -241,20 +240,17 @@ config("default_cpp_standard") {
...
+      # These flags apply to non-modular windows platforms which includes win32, xb1.
+      configs = [ ":win_cpp_config" ]
```
Comment (andrewsavage1) [inline starboard/build/config/BUILD.gn:245]: I think this is actually less readable given that it's done for some platforms but not others and also only done for the c++17 config but not the 20 config. I'd just get rid of it and have the flags inline, but if you want to keep it let's make it consistent

**Review by andrewsavage1 — APPROVED** [review]

File: starboard/build/config/BUILD.gn (hunk 2 above)
Comment (niranjanyardi) [reply starboard/build/config/BUILD.gn:245]: i want the files in a single config - as they were always like that before i refactored them in a single place and spliut the flags into host and default configs.

Ill rename `win_cpp_config` to `win_cpp_17_config` and add a `win_cpp_20_config` to amek it consistent

File: starboard/build/config/BUILD.gn (hunk 1 above)
Comment (niranjanyardi) [reply starboard/build/config/BUILD.gn:243]: done

**Pull request merged**